src/Migrations/M034MigrateCoversContentBlockToPostsListBlock.php

```diff
@@ -0,0 +1,465 @@
+            );
+
+            // If there are no posts, abort.
+            if (!$posts) {
+                return;
+            }
+
+            echo "Covers content block migration in progress...\n"; // phpcs:ignore
+
+            $parser = new WP_Block_Parser();
+
+            foreach ($posts as $post) {
+                if (empty($post->post_content)) {
+                    continue;
+                }
+
+                $current_post_id = $post->ID; // Store the current post ID
+
+                echo 'Parsing post ', $current_post_id, "\n"; // phpcs:ignore
+
+                // Get all the blocks of each post.
+                $blocks = $parser->parse($post->post_content);
+
+                if (!is_array($blocks)) {
+                    throw new \Exception("Invalid block structure for post #" . $current_post_id);
+                }
+
+                foreach ($blocks as &$block) {
+                    // Check if the block is valid.
+                    if (!is_array($block)) {
+                        continue;
+                    }
+
+                    // Check if the block has a 'blockName' key.
+                    if (!isset($block['blockName'])) {
+                        continue;
+                    }
+
+                    // Check if the block is a Cover block. If not, abort.
+                    if ($block['blockName'] !== Utils\Constants::BLOCK_COVERS) {
+                        continue;
+                    }
+
+                    // Check if the Cover block type is Content. If not, abort.
+                    // Cover blocks of type content have as value of $block['attrs']['cover_type']
+                    // the following possibilities: "content", "1", NULL
+                    // https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/26b480a0954667a0813ca8c3a90377f2a1fbea1c/classes/blocks/class-covers.php#L35
+                    // https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/26b480a0954667a0813ca8c3a90377f2a1fbea1c/classes/blocks/class-covers.php#L191
+                    $type = $block['attrs']['cover_type'];
+                    $const = Utils\Constants::COVER_BLOCK_TYPES['content'];
+
+                    // phpcs:disable Use early exit to reduce code nesting
+                    if (!isset($type) || $type === $const['name'] || $type === $const['number']) {
+                        // Get the block attributes.
+                        $attrs = self::get_posts_list_block_attrs($block);
+
+                        // Transform the cover block into a posts list block.
+                        $block = self::create_query_block($attrs);
+                    }
+                    // phpcs:enable Use early exit to reduce code nesting
+                }
+
+                // Unset the reference to prevent potential issues.
+                unset($block);
+
+                // Serialize the blocks content.
+                $new_content = serialize_blocks($blocks);
```

Comment:
We should add a condition here to check if the post content has changed. Otherwise, for all non-content covers, the update query will execute unnecessarily and display a successful migration message even though no migration has occurred.
 ```
 if ($post->post_content === $new_content) {
        continue;
 }
```

Reply:
Done!